packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection.tsx

```diff
@@ -0,0 +1,260 @@
+
+  const isLocalhost =
+    connectionInfo.connectionOptions.connectionString.startsWith(
+      'mongodb://localhost'
+    );
+
+  const maybeProtectConnectionString = useMaybeProtectConnectionString();
+  const [hoverProps, isHovered] = useHoverState();
+  const isFavorite = connectionInfo.savedConnectionType === 'favorite';
+  const { openToast } = useToast('compass-connections');
+
+  const icon = isLocalhost ? (
+    <Icon className={iconStyles} glyph="Laptop" />
+  ) : isFavorite ? (
+    <Icon className={iconStyles} glyph="Favorite" />
+  ) : (
+    SERVER_ICON
+  );
+
+  async function copyConnectionString(connectionString: string) {
+    try {
+      await navigator.clipboard.writeText(connectionString);
+      openToast('copy-to-clipboard', {
+        title: 'Success',
+        description: 'Copied to clipboard.',
+        variant: 'success',
+        timeout: TOAST_TIMEOUT_MS,
+      });
+    } catch (err) {
+      openToast('copy-to-clipboard', {
+        title: 'Error',
+        description:
+          'An error occurred when copying to clipboard. Please try again.',
+        variant: 'warning',
+        timeout: TOAST_TIMEOUT_MS,
+      });
+    }
+  }
+
+  const actions: (ItemAction<Action> | ItemSeparator)[] = [
+    {
+      action: 'edit-connection',
+      label: 'Edit connection',
+      icon: 'Edit',
+    },
+    {
+      action: 'copy-connection-string',
+      label: 'Copy connection string',
+      icon: 'Copy',
+    },
+    {
+      action: 'toggle-favorite',
+      label:
+        connectionInfo.savedConnectionType === 'favorite'
+          ? 'Unfavorite'
+          : 'Favorite',
+      icon: 'Favorite',
+    },
+    {
+      action: 'duplicate-connection',
+      label: 'Duplicate',
+      icon: 'Clone',
+    },
+    { separator: true },
+    {
+      action: 'remove-connection',
+      label: 'Remove',
+      icon: 'Trash',
+      variant: 'destructive',
+    },
+  ];
+
+  const onAction = useCallback(
+    (action: Action) => {
+      switch (action) {
+        case 'edit-connection':
+          return onEditConnection(connectionInfo);
+        case 'copy-connection-string':
+          return void copyConnectionString(
+            maybeProtectConnectionString(
+              connectionInfo.connectionOptions.connectionString
+            )
+          );
+        case 'toggle-favorite':
+          return onToggleFavoriteConnection(connectionInfo);
+        case 'duplicate-connection':
+          return onDuplicateConnection(connectionInfo);
+        case 'remove-connection':
+          return onDeleteConnection(connectionInfo);
+      }
+    },
+    [
+      connectionInfo,
+      onEditConnection,
+      copyConnectionString,
+      maybeProtectConnectionString,
+      onToggleFavoriteConnection,
+      onDuplicateConnection,
+      onDeleteConnection,
+    ]
+  );
+  return (
+    <li
+      {...hoverProps}
+      style={{
+        backgroundColor: connectionColorToHex(connectionInfo.favorite?.color),
```

Comment:
The background colour will be different in case of light/dark theme and they are designed for accessibility.
Right now it's partially implemented, but it will be finished in COMPASS-7717:

https://jira.mongodb.org/browse/COMPASS-7717

Reply:
Ah, so it's not implemented yet, got it